### PR TITLE
Add support to run on Kubernetes, add Helm chart

### PR DESCRIPTION
**Pull request by leonklingele**
### TODOs

- [x] Add official Docker image (#977)
- [x] Make Invidious configurable (#1016)
- [x] Add resource constraints
- [x] Add readiness and liveness probe

### CODE REVIEW

File: kubernetes/values.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: invidious
+
+image:
+  repository: invidious/invidious
```
Comment (leonklingele) [inline kubernetes/values.yaml]: This needs to point to the official `invidious` Docker image. See #977.

**Comment by leonklingele** [comment]
@omarroth any idea on how to configure Invidious without a config file (e.g. by env vars)? Using config files doesn't really work inside Kubernetes.

**Comment by unixfox** [comment]
@leonklingele it works, I have seen for example a guy do it for Searx here: https://github.com/travnewmatic/k8s-searx/blob/master/searx.yaml#L240

**Comment by leonklingele** [comment]
The only thing missing is the Docker image :)

**Comment by leonklingele** [comment]
For anyone curious about trying this: Simply use `tag: latest-k8s` inside the Helm configuration file (`values.yaml`). It's a Docker image built from 9841f74adc94a4845aea6aace8d2408c3255dfa7 with #1016 applied.

**Comment by leonklingele** [comment]
This is no longer work in progress and ready to be reviewed / merged :)

Everything is running smoothly on my 4-node cluster.

**Comment by omarroth** [comment]
Looks great! Much appreciated :)

**Pull request merged**